pkg/kubelet/util/manager/cache_based_manager.go

```diff
@@ -230,6 +235,10 @@ func (c *cacheBasedManager) RegisterPod(pod *v1.Pod) {
 	prev = c.registeredPods[key]
 	c.registeredPods[key] = pod
 	if prev != nil {
+		if prev.Namespace != pod.Namespace || prev.Name != pod.Name {
+			panic(fmt.Errorf("236: mismatch in key/obj: %#v, %#v", key, prev))
+		}
+		klog.InfoS("RegisterPod.prev", "groupResource", c.groupResource, "namespace", prev.Namespace, "name", prev.Name)
```

Comment:
@liggitt - thanks for opening this PR - I think it would be useful to have this PR merged, though I'm wondering if we don't want to decrease the verbosity of these logs.